rotkehlchen/chain/evm/decoding/balancer/mixins.py

```diff
@@ -0,0 +1,111 @@
+from typing import TYPE_CHECKING
+
+from rotkehlchen.chain.evm.decoding.balancer.constants import CPT_BALANCER_V1, CPT_BALANCER_V2
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import EvmTransaction
+
+if TYPE_CHECKING:
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+
+
+class BalancerCommonAccountingMixin:
+    def __init__(self, counterparty: str) -> None:
```

Comment:
Write out the possible types

```suggestion
    def __init__(self, counterparty: Literal['choice1', 'choice2']) -> None:
```